applications/RomApplication/costum_strategies/rom_builder_and_solver.h

```diff
@@ -0,0 +1,811 @@
+                {
+                    //calculate elemental contribution
+                    pScheme->CalculateSystemContributions(*(it.base()), LHS_Contribution, RHS_Contribution, EquationId, CurrentProcessInfo);
+					Element::DofsVectorType dofs;
+					it->GetDofList(dofs, CurrentProcessInfo);
+                    //assemble the elemental contribution - here is where the ROM acts
+                    //compute the elemental reduction matrix T
+                    const auto& geom = it->GetGeometry();
+                    Matrix Telemental(geom.size()*mNodalDofs, mRomDofs);
+					
+                    for(unsigned int i=0; i<geom.size(); ++i)
+                    {
+                        const Matrix& rom_nodal_basis = geom[i].GetValue(ROM_BASIS);
+                        for(unsigned int k=0; k<rom_nodal_basis.size1(); ++k)
+                        {
+							if (dofs[i*mNodalDofs + k]->IsFixed())
+								row(Telemental, i*mNodalDofs + k) = ZeroVector(Telemental.size2());
+							else
+								row(Telemental, i*mNodalDofs+k) = row(rom_nodal_basis,k);
+						}
+
+                    }
+
+                    Matrix aux = prod(LHS_Contribution, Telemental);
+					noalias(Arom) += prod(trans(Telemental), aux);
+                    noalias(brom) += prod(trans(Telemental), RHS_Contribution);
+
+                    // clean local elemental me overridemory
+                    pScheme->CleanMemory(*(it.base()));
+                }  
+            } 
+
+ 
+            // #pragma omp for  schedule(guided , 512)
+            for (int k = 0; k < nconditions;  k++)
+            {
+                ModelPart::ConditionsContainerType::iterator it = cond_begin + k;
+
+                //detect if the element is active or not. If the user did not make any choice the element
+                //is active by default
+                bool condition_is_active = true;
+                if ((it)->IsDefined(ACTIVE))
+                    condition_is_active = (it)->Is(ACTIVE);
+
+                if (condition_is_active)
+                {
+                    Condition::DofsVectorType dofs;
+					it->GetDofList(dofs, CurrentProcessInfo);    
+                    //calculate elemental contribution
+                    pScheme->Condition_CalculateSystemContributions(*(it.base()), LHS_Contribution, RHS_Contribution, EquationId, CurrentProcessInfo);
+
+                    //assemble the elemental contribution - here is where the ROM acts
+                    //compute the elemental reduction matrix T
+                    const auto& geom = it->GetGeometry();
+                    Matrix Telemental(geom.size()*mNodalDofs, mRomDofs);
+
+                    for(unsigned int i=0; i<geom.size(); ++i)
+                    {
+                        const Matrix& rom_nodal_basis = geom[i].GetValue(ROM_BASIS);
+                        for(unsigned int k=0; k<rom_nodal_basis.size1(); ++k)
+                        {
+							if (dofs[i*mNodalDofs + k]->IsFixed())
```

Comment:
```suggestion
        row(Telemental, i*mNodalDofs + k) = dofs[i*mNodalDofs + k]->IsFixed() ? ZeroVector(Telemental.size2()) : row(rom_nodal_basis,k);
```
I think that you can substitute this if statement by a unique ternary operator.